lib/project_types/extension/models/specification_handlers/checkout_ui_extension.rb

```diff
@@ -22,6 +33,74 @@ def build_resource_url(context:, shop:)
           return unless product
           format("/cart/%<variant_id>d:%<quantity>d", variant_id: product.variant_id, quantity: 1)
         end
+
+        private
+
+        def localization(context)
+          Dir.chdir(context.root) do
+            locale_filenames = Dir["**/*"].select { |filename| valid_l10n_file?(filename) }
```

Comment:
Let me adjust my previous feedback. I missed that you were doing a global directory search for file names. However, I thought that we were going to be keeping locale files in the `locales/` directory. So, instead, can we change this to be only searching for files in `Dir["locales"`]` and error if any files are invalid?

Reply:
We are indeed only taking `.json` files under `locales` and erroring out on invalid file types (e.g. `txt`) or names (e.g. `french.json` instead of `fr.json`).
I had tried using a more specific `Dir` statement to take only the `locales` dir to begin with but was running into some issues. I can look into this again now that the rest of this is more stable.